internal/venice-common/src/main/java/com/linkedin/venice/pubsub/adapter/kafka/TopicPartitionsOffsetsTracker.java

```diff
@@ -64,51 +55,81 @@ public TopicPartitionsOffsetsTracker(@Nonnull Duration offsetsUpdateInterval, Du
    * to be the largest offset amongst the records passed in.
    *
    * @param records consumed records
+   * @param kafkaConsumer from which to extract metrics
    */
   public void updateEndAndCurrentOffsets(
       ConsumerRecords<byte[], byte[]> records,
-      Map<MetricName, ? extends Metric> metrics) {
-    if (lastMetricsCollectedTime != null && LatencyUtils
-        .getElapsedTimeInMs(lastMetricsCollectedTime.toEpochMilli()) < offsetsUpdateInterval.toMillis()) {
+      Consumer<byte[], byte[]> kafkaConsumer) {
+    long lastMetricsCollectedTimeSnapshot = lastMetricsCollectedTime;
+    long currentTime = System.currentTimeMillis();
+    long elapsedTime = currentTime - lastMetricsCollectedTimeSnapshot;
+    if (elapsedTime < offsetsUpdateIntervalMs) {
```

Comment:
Since offsets are fetched from metrics IMO the cost is relatively trivial to update offsets upon each poll request. Can we consider removing this check altogether?

Reply:
Hmm... possibly, yeah. I guess that would depend on a few factors, including:

1. What is the batch size? If the batch size is small (in the most extreme case, containing just a single record), then we would pay this overhead in a less amortized way.
2. How many metrics are there in the client? My understanding is that Kafka clients carry a fairly large number of metrics, so iterating over each one and comparing the name for each might actually be costly... but there may be another algorithm that is more efficient, e.g. hanging on to the `KafkaMetric` object itself rather than copying its content...

So I guess we could expose the param to tune the time interval, and try lowering it to see if the cost is noticeable. I could remove the safeguard against making the interval zero, since you pointed out that it is single threaded anyway.

Reply:
Good point. I agree it would be better to make interval tunable instead of removing it completely. We can do it in follow-up PR as well. Thanks

Reply:
As per offline discussion we will take up this in a follow up PR